core/trino-main/src/main/java/io/trino/execution/CreateTableTask.java

```diff
@@ -166,15 +167,19 @@ public String explain(CreateTable statement, List<Expression> parameters)
             }
             else if (element instanceof LikeClause) {
                 LikeClause likeClause = (LikeClause) element;
-                QualifiedObjectName likeTableName = createQualifiedObjectName(session, statement, likeClause.getTableName());
-                if (metadata.getCatalogHandle(session, likeTableName.getCatalogName()).isEmpty()) {
-                    throw semanticException(CATALOG_NOT_FOUND, statement, "LIKE table catalog '%s' does not exist", likeTableName.getCatalogName());
+                QualifiedObjectName originalLikeTableName = createQualifiedObjectName(session, statement, likeClause.getTableName());
+                if (metadata.getCatalogHandle(session, originalLikeTableName.getCatalogName()).isEmpty()) {
+                    throw semanticException(CATALOG_NOT_FOUND, statement, "LIKE table catalog '%s' does not exist", originalLikeTableName.getCatalogName());
                 }
+
+                TableHandle likeTable = metadata.getRedirectedTableHandle(session, originalLikeTableName)
+                        .orElseThrow(() -> semanticException(TABLE_NOT_FOUND, statement, "LIKE table '%s' does not exist", originalLikeTableName));
+
+                QualifiedObjectName likeTableName = metadata.getRedirectedTableName(session, originalLikeTableName);
                 if (!tableName.getCatalogName().equals(likeTableName.getCatalogName())) {
-                    throw semanticException(NOT_SUPPORTED, statement, "LIKE table across catalogs is not supported");
+                    String redirectionMessage = originalLikeTableName.equals(likeTableName) ? "" : format("(LIKE table '%s' has been redirected to '%s'.)", originalLikeTableName, likeTableName);
+                    throw semanticException(NOT_SUPPORTED, statement, "LIKE table across catalogs is not supported. %s", redirectionMessage);
```

Comment:
This will end the message with a space when the message is empty. How about
```java
String message = "CREATE TABLE LIKE across catalogs is not supported";
if (!originalLikeTableName.equals(likeTableName)) {
    message += format(". LIKE table '%s' redirected to '%s'.", originalLikeTableName, likeTableName);
}
```
This also uses the "table redirected to" present tense.